pkg/client/profile/jwt_profile.go

```diff
@@ -85,9 +104,13 @@ func (j *jwtProfileTokenSource) HttpClient() *http.Client {
 }
 
 func (j *jwtProfileTokenSource) Token() (*oauth2.Token, error) {
```

Comment:
Wouldn't a better option be to make the context a parameter here too, so that you don't have to create a context.Background in the function? As this function is used to create a request.

Reply:
That was my first intention. However, `Token()` appears to implement the [`oauth2.TokenSource`](https://pkg.go.dev/golang.org/x/oauth2#TokenSource). Therefore I created a local interface that embeds the `TokenSource` and extends it with `TokenCtx()` to allow for calls with context.

https://github.com/zitadel/oidc/blob/6d455501aed493f3d0c8ce3a364ecf1db5f6a306/pkg/client/profile/jwt_profile.go#L15-L18

Another option would be to drop the `oauth2.TokenSource` compatibility, but I'm not sure if that breaks users that specifically need it.